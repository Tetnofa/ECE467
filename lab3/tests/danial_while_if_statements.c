int main() {
    int x;
    x = 1;
    if (x == 1) {
        if (x == 1) {
            int y;
            y = 1;
            if (y != 1) {

            } else {
                if (y <= 2) {
                    if (z) {
                        if (z) {

                        }
                    }
                }
            }
        }
    }
}
void poo () {
    if (1 < x) {

    }
}
int a (int b, int c) {
    if (b < c) {
        if (x) {

        }else {
            if (b != c) {

            }
        }
    }
}
int f (int arr[],int b, int c) {
    if (arr[3] <= b) {
        f(arr, b, c);
    } else if (arr[3] >= c) {
        f(arr, b, c);
    } else if (arr[5] > y) {
        poo();
    }
}
void fcn() {
    while (1 < 2) {

    }
    while (x) {

    }
    while (y) {
        while ( 1 < 2) {
            while (x) {
                if (f) {
                    if (1 == 2) {
                        while (1 != 8) {
                            if (5==3) {
                                if (5 <= 3) {
                                    while (3) {

                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}

void whileScope() {
    while (1) {
        int x;
        x = 2;
        if (x == 2) {
            int y;
        }
    }
}
void pee() {
    int x;
    while (1) {
        int y;
        y = 9;
        if (x <= y) {
            if (x >= 9) {
                if (x != y) {

                }
            }
        }
    }
}
bool fadwa() {
    bool x;
    bool y;
    while (x) {
        if (y) {

        }
    }
}