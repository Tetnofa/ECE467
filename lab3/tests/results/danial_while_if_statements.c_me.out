sema: 12:24 : use of undefined identifier z
sema: 13:28 : use of undefined identifier z
sema: 13:24 : invalid condition in if statement
sema: 12:20 : invalid condition in if statement
sema: 23:12 : use of undefined identifier x
sema: 23:8 : type mismatch 
sema: 23:4 : invalid condition in if statement
sema: 29:12 : use of undefined identifier x
sema: 29:8 : invalid condition in if statement
sema: 43:24 : use of undefined identifier y
sema: 43:15 : type mismatch 
sema: 43:11 : invalid condition in if statement
sema: 51:11 : use of undefined identifier x
sema: 51:4 : invalid condition in while statement
sema: 54:11 : use of undefined identifier y
sema: 56:19 : use of undefined identifier x
sema: 57:20 : use of undefined identifier f
sema: 62:36 : invalid condition in while statement
sema: 57:16 : invalid condition in if statement
sema: 56:12 : invalid condition in while statement
sema: 54:4 : invalid condition in while statement
sema: 76:4 : invalid condition in while statement
sema: 86:4 : invalid condition in while statement
